File: MATLAB Test Scripts/atmosphere_test.m
function atmosphere_test
%{
    Tests the atmosphere function by plotting the density
    as a function of altitude from sea level through 1000 km.
%}

%...Geometric altitudes (km):
z = linspace(0, 1000, 500); % Generate 500 points from 0 to 1000 km

%...Calculate densities:
density = arrayfun(@atmosphere, z);

%...Plot the results:
figure;
semilogy(z, density, 'LineWidth', 1.5);
grid on;
title('Atmospheric Density vs. Altitude');
xlabel('Altitude (km)');
ylabel('Density (kg/m^3)');
legend('Density', 'Location', 'northeast');

end

File: tests/cpp/Sources/atmosphere_test.cpp
#include <iostream>
#include <vector>
#include <cmath>
#include <iomanip>
#include "atmosphere.h"

/*
    Tests the atmosphere function by plotting the density
    as a function of altitude from sea level through 1000 km.
*/
int main() {
    //...Geometric altitudes (km):
    const int num_points = 500;
    const double z_min = 0.0;
    const double z_max = 1000.0;
    std::vector<double> z(num_points);
    std::vector<double> density(num_points);

    for (int i = 0; i < num_points; ++i) {
        z[i] = z_min + i * (z_max - z_min) / (num_points - 1);
    }

    //...Calculate densities:
    for (int i = 0; i < num_points; ++i) {
        density[i] = atmosphere(z[i]);
    }

    //...Output the results:
    std::cout << std::fixed << std::setprecision(15);
    std::cout << "Altitude (km)\tDensity (kg/m^3)\n";
    std::cout << "--------------------------------\n";
    for (int i = 0; i < num_points; ++i) {
        std::cout << z[i] << "\t" << density[i] << "\n";
    }

    return 0;
}
